src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -185,9 +185,10 @@ internal RawRecord ReadRaw()
             int serializedEventLength = ReadInt32();
             Stream stream = _binaryReader.BaseStream.Slice(serializedEventLength);
 
-            _lastSubStream = stream as SubStream;
-
-            _recordNumber += 1;
+            using (_lastSubStream = stream as SubStream)
```

Comment:
Is it ok to dispose this while keeping it in the field `_lastSubStream`? That doesn't seem right to me (without looking hard).